src/js/core/util/scrolling.js

```diff
@@ -384,11 +384,32 @@
 				}
 			}
 
+			/**
+			 * Check visible state of the element
+			 * @param {Element} elm
+			 */
+			function _isVisible(elm) {
+				const style = getComputedStyle(elm);
```

Comment:
In this case for checking the visibility of element enough elm.getBoundingClientRect() and check height property.